WcaOnRails/app/webpacker/components/Media/MediaTable.jsx

```diff
@@ -0,0 +1,257 @@
+import React, { useEffect, useState } from 'react';
+
+import _ from 'lodash';
+import { Table, Confirm, Modal, Button } from 'semantic-ui-react';
+import { competitionUrl, apiV0Urls } from '../../lib/requests/routes.js.erb';
+import { countries, continents, years } from '../../lib/wca-data.js.erb';
+import { dateRangeBetween } from '../../lib/helpers/media-table';
+import useLoadedData from '../../lib/hooks/useLoadedData';
+import useSaveAction from '../../lib/hooks/useSaveAction';
+import {
+  DropdownMenu,
+  DropdownItem,
+  DropdownHeader,
+  DropdownDivider,
+  Dropdown,
+} from 'semantic-ui-react'
+export default function MediaTable({ isValidate }) {
+  const countryOptions = _.map(countries.byIso2, (country) => ({
+    key: country.iso2,
+    text: country.name,
+    value: country.iso2,
+  }));
+  const continentOptions = continents.map((continent) => ({
+    key: continent.id,
+    text: continent.name,
+    value: continent.name,
+  }));
+  const yearOptions = [
+    {
+      key: 'all',
+      text: 'All Years',
+      value: 'All Years',
+    },
+    ...years.map((year) => ({
+      key: year,
+      text: year,
+      value: year,
+    })),
+  ];
+
+
+  const handleUpdateSuccess = () => {
+    setModalOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setModalOpen(false);
+  };
+  const [selectedYear, setSelectedYear] = useState(yearOptions[1].value);
+  const [selectedRegion, setSelectedRegion] = useState("All Regions");
+  const { save, saving } = useSaveAction();
+  const [mediaCombined, setMediaCombined] = useState([])
+  const type = isValidate ? 'pending' : 'accepted'
+  const { loading, error, data, sync } = useLoadedData(
+    apiV0Urls.media.list(type, selectedYear, selectedRegion)
+  );
+  const [modalOpen, setModalOpen] = useState(false);
+
+  useEffect(() => {
+    if (data) {
+      setMediaCombined(data?.map((medium, idx) => {
```

Comment:
The current implementation isn't necessarily bad, but generally avoiding side effects and having fewer states is good. I'd suggest making `mediaCombined` state-less:
```
const mediaCombined = data?.map((medium, idx) => { ... })
```
and then having a separate (state-less, memoized) const for the sorted data:
```
const sortedMediaCombined = useMemo(
  () => [...mediaCombined]?.sort(...), // probably doesn't work exactly as written in the null case
  [mediaCombined, sortColumn, sortDirection],
]
```
and then using
```
{sortedMediaCombined?.map((media_row) => (
```
in the jsx return.

Thoughts?